src/engine/x86-64/X86_64MacroAssembler.v3

```diff
@@ -963,6 +961,11 @@ class X86_64MacroAssembler extends MacroAssembler {
 		asm.movq_s_r(scratch, src);
 		asm.movddup_s_s(dst, scratch);
 	}
+	def emit_i8x16_swizzle(dst: X86_64Xmmr, src: X86_64Xmmr, tmp: X86_64Gpr, mask: X86_64Xmmr) {
```

Comment:
Does this implementation handle out-of-bounds cases properly? 

(https://emscripten.org/docs/porting/simd.html?highlight=exception)

Reply:
For 2, actually the cases are handled but not covered by tests. Adding new tests